object/provider.go

```diff
@@ -225,7 +225,6 @@ func UpdateProvider(id string, provider *Provider) (bool, error) {
 		session = session.Omit("client_secret2")
 	}
 
-	provider.Endpoint = util.GetEndPoint(provider.Endpoint)
 	provider.IntranetEndpoint = util.GetEndPoint(provider.IntranetEndpoint)
```

Comment:
The codes here are about https://github.com/casdoor/casdoor/commit/96219442f57838324e9d9efff701669b6ca97142
Now it  looks like the pr breaks other provider such as SAML. Maybe you need to add an if condition to prevent the code breaks the fix in the https://github.com/casdoor/casdoor/commit/96219442f57838324e9d9efff701669b6ca97142

Reply:
@XDTD The pr works, but we need to compatible the fix in https://github.com/casdoor/casdoor/commit/96219442f57838324e9d9efff701669b6ca97142

Reply:
Thx for ur advice. This pr has been updated. Pls review the changes.